enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/IdentityVerificationPrecompleteGuard.java

```diff
@@ -0,0 +1,173 @@
+                .map(DocumentVerificationEntity::getStatus)
+                .allMatch(it -> it == DocumentStatus.ACCEPTED)) {
+            logger.debug("Some documents are not accepted for identity verification ID: {}", idVerification.getProcessId());
+            return Result.failed("Some documents not accepted");
+        }
+
+        if (!requiredDocumentTypesGuard.evaluate(documentVerifications, idVerification.getId())) {
+            logger.debug("Not all required documents are present for verification ID: {}", idVerification.getProcessId());
+            return Result.failed("Required documents not present");
+        }
+
+        if (!isPrefinalPhaseAndStateValid(idVerification)) {
+            logger.debug("Not valid phase and state for verification ID: {}", idVerification.getProcessId());
+            return Result.failed("Not valid phase and state");
+        }
+
+        if (!isOtpValid(idVerification)) {
+            logger.debug("Not valid OTP for verification ID: {}", idVerification.getProcessId());
+            return Result.failed("Not valid OTP");
+        }
+
+        if (!isPresenceCheckValid(idVerification)) {
+            logger.debug("Presence check did not pass for verification ID: {}", idVerification.getProcessId());
+            return Result.failed("Presence check did not pass");
+        }
+
+        if (!isActivationValid(idVerification)) {
+            logger.debug("Activation is not valid for verification ID: {}", idVerification.getProcessId());
+            return Result.failed("Activation is not valid");
+        }
+
+        return Result.successful();
+    }
+
+    private boolean isActivationValid(IdentityVerificationEntity idVerification) throws RemoteCommunicationException {
+        final ActivationStatus activationStatus = activationService.fetchActivationStatus(idVerification.getActivationId());
+        return activationStatus == ActivationStatus.ACTIVE;
+    }
+
+    private boolean isOtpValid(final IdentityVerificationEntity idVerification) {
+        if (!identityVerificationConfig.isVerificationOtpEnabled()) {
+            logger.trace("OTP verification is disabled");
+            return true;
+        }
+        return onboardingOtpRepository.findNewestByProcessIdAndType(idVerification.getProcessId(), OtpType.USER_VERIFICATION)
+                .map(OnboardingOtpEntity::getStatus)
+                .filter(it -> it == OtpStatus.VERIFIED)
+                .isPresent();
+    }
+
+    private boolean isPresenceCheckValid(final IdentityVerificationEntity identityVerification) {
+        if (!identityVerificationConfig.isPresenceCheckEnabled()) {
+            logger.trace("Presence check is disabled");
+            return true;
+        }
+        final RejectOrigin rejectOrigin = identityVerification.getRejectOrigin();
+        final ErrorOrigin errorOrigin = identityVerification.getErrorOrigin();
+
+        return errorOrigin != ErrorOrigin.PRESENCE_CHECK && rejectOrigin != RejectOrigin.PRESENCE_CHECK;
```

Comment:
This seems strange. Why would we allow any error origins or reject origins when the identity verification succeeds? Maybe this is a flaw in current logic, that the error is not reset when the step succeeds?

Ideally, there should rather be a check that there are no errors or rejections present.

Reply:
Ok, let's remove it until #453